src/firefox/ffPage.ts

```diff
@@ -86,13 +86,15 @@ export class FFPage implements PageDelegate {
   }
 
   async _initialize() {
+    const geolocation = this._browserContext._options.geolocation;
     try {
       await Promise.all([
         // TODO: we should get rid of this call to resolve before any early events arrive, e.g. dialogs.
         this._session.send('Page.addScriptToEvaluateOnNewDocument', {
           script: '',
           worldName: UTILITY_WORLD_NAME,
         }),
+        geolocation ? this._setGeolocation(geolocation) : Promise.resolve(),
```

Comment:
This makes popups not have geolocation override at the start. We should push to the browser context instead.